File: src/main/java/com/example/demoimdb/model/Movie.java
package com.example.demoimdb.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;
import org.hibernate.validator.constraints.Length;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "movie")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class Movie {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(name = "name")
    private String name;
    @Column(name = "description", length = 1000)
    private String description;
    @Column(name = "image")
    private String image;
    @Column(name = "trailer")
    private String trailer;
    @Column(name = "releaseDate")
    private String releaseDate;
    @Column(name = "duration")
    private Integer duration;
    @Column(name = "type")
    private Integer type;
    @Column(name = "endYear")
    private Integer endYear;
    @Column(name = "number_season")
    private Integer numberSeason;
    @Column(name = "number_vote")
    private Integer numberVote;
    @Column(name = "score")
    private Double score;
    @Column(name = "language")
    private String language;
    @OneToMany(mappedBy = "movie")
    private List<MovieGenre> listMovieGenres;
    @OneToMany(mappedBy = "movie")
    private List<MovieActor> listMovieActors;
    @OneToMany(mappedBy = "movie")
    private List<MovieDirector> listMovieDirectors;
    @OneToMany(mappedBy = "movie")
    private List<Comment> listComments;
    @JsonIgnore
    @OneToMany(mappedBy = "movie")
    private List<Rating> listRatings;
    @OneToMany(mappedBy = "movie")
    private List<Episode> listEpisodes;
}

File: src/entities/Movie.ts
import { Column, Entity, OneToMany, PrimaryGeneratedColumn } from 'typeorm';
import { MovieGenre } from './MovieGenre';
import { Rating } from './Rating';
import { Comment } from './Comment';
import { MovieActor } from './MovieActor';
import { MovieDirector } from './MovieDirector';
import { Episode } from './Episode';

@Entity({ name: 'movies' })
export class Movie {
  @PrimaryGeneratedColumn()
  id: number;

  @Column()
  name: string;

  @Column()
  type: number;

  @Column({ nullable: true, length: 1000 })
  description: string;

  @Column({ nullable: true })
  trailer: string;

  @Column({ nullable: true })
  image: string;

  @Column({ nullable: true })
  releaseDate: string;

  @Column({ nullable: true })
  duration: number;

  @Column({ nullable: true })
  ended: boolean;

  @Column({ nullable: true })
  numberSeason: number;

  @Column({ nullable: true })
  numberVote: number;

  @Column({ nullable: true, type: 'float' })
  score: number;

  @Column({ nullable: true })
  language: string;

  @OneToMany(() => Episode, (episode) => episode.movie)
  episodes: Episode[];

  @OneToMany(() => MovieActor, (movieActor) => movieActor.movie)
  movieActors: MovieActor[];

  @OneToMany(() => MovieDirector, (movieDirector) => movieDirector.movie)
  movieDirectors: MovieDirector[];

  @OneToMany(() => MovieGenre, (movieGenre) => movieGenre.movie)
  movieGenres: MovieGenre[];

  @OneToMany(() => Rating, (rating) => rating.movie)
  ratings: Rating[];

  @OneToMany(() => Comment, (comment) => comment.movie)
  comments: Comment[];
}
